newrelic/resource_newrelic_alert_channel.go

```diff
@@ -141,6 +173,19 @@ func resourceNewRelicAlertChannelRead(d *schema.ResourceData, meta interface{})
 
 	d.Set("name", channel.Name)
 	d.Set("type", channel.Type)
+
+	// extract headers from Configuration before we try and set it in the resource
+	if headers, ok := channel.Configuration["headers"]; ok {
+		d.Set("headers", headers)
+		delete(channel.Configuration, "headers")
```

Comment:
I'm a little hesitant to do this `delete` as its a breaking change for anyone already using this behavior.

Reply:
I do not believe that the `delete` could break anything because, as far as I can tell setting the headers or payload is broken currently

Reply:
I think nathanael-ness is right, I don't see how either `headers` or `payload` could be working currently with the type mismatch.

We're having to use a null-resource to deploy a webhook alert channel, can't use the resource as it is.